src/sync/chase_lev.rs

```diff
@@ -158,69 +155,93 @@ impl<T> Deque<T> {
     unsafe fn push(&self, data: T) {
         let guard = epoch::pin();
 
-        let mut b = self.bottom.load(SeqCst);
-        let t = self.top.load(SeqCst);
-        let mut a = self.array.load(SeqCst, &guard).unwrap();
-        let size = b - t;
+        let mut b = self.bottom.load(Relaxed);
+        let t = self.top.load(Acquire);
+        let mut a = self.array.load(Relaxed, &guard).unwrap();
+        
+        let size = b.wrapping_sub(t);
```

Comment:
Hm ok, let's stick with panicking operations for now and defer to a time where we can analyze a little more in detail about whether a wrapping operation is needed or not. I'm fine doing this in a follow-up PR though!